ursine/header_parsing.py

```diff
@@ -0,0 +1,58 @@
+'''Parsing for SIP URIs.'''
+from .uri import URI
+from collections import namedtuple
+
+
+HeaderParseResult = namedtuple('HeaderParseResult', (
+    'display_name',
+    'parameters',
+    'uri',
+))
+
+
+def parse_params(params_str):
+    params = {}
```

Comment:
Can params be duplicated?

Reply:
Nope

Reply:
KK. (For what its worth, you could just yield tuples then).

Reply:
Could do, but I figured since making the dictionary is required anyway even if it's maybe a little bit spaghettied it'd be more performant to parse right into the dictionary and it can be used directly by the resultant header object.